### DISCUSSION TITLE
Enabling Messenger in console

### DISCUSSION THREAD

**Discussion by papaye1855** [PrestaShop/PrestaShop#36661] [Ideas]
### The problem

I created a plugin that allows me to import products from an external API. The problem that I had was that after getting the json response from the API, when I looped to create the products in Prestashop, it took too long because it was done on the same HTTP process and as a result I had 500 errors after a while and it was only a few products that were created but not all of them.

It is with this in mind that I wanted to do the task of creating the products in the background. I did what was necessary on the code side to be able to integrate symfony Messenger and dispatch the message

```
private function handleFormSubmission($categoryId) {
        $categoryName = $this->getCategoryNameById($categoryId);
        $products = $this->fetchProductsFromApi($categoryName);

        // Here we dispatch the symfony/messenger event so it can be handled in background
        $this->messageBus->dispatch(new ImportProductMessage($products, $categoryId));
 }
```
**It is from this moment that the real problem begins**, because after dispatching the event, **we must tell messenger to consume the tasks in the background with the command**:

```
   php bin/console messenger:consume async
```

![Issue Messenger](https://github.com/user-attachments/assets/718cf261-8f0c-42db-863f-bc12395bfb5d)

but there is no messenger command with the console but yet Messenger is in the symfony components of Prestashop 8.X
Therefore it is practically impossible to consume background tasks asynchronously with messenger.

### The solution

It would be nice to be able to have messenger in the console for useful cases like this.

### Additional context

And I will be delighted to learn from your expertise and experiences how to resolve a problem like this if you already encounter one.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @papaye1855 

Symfony Messenger has been added to PrestaShop 9 (still in development). You'd need to add it as a dependency to your module.
